### PR TITLE
Configure techdocs-publish action to work with workload identity

### PR DESCRIPTION
**Pull request by IuryAlves**
Ticket: FLAT-2085

### CODE REVIEW

File: techdocs-publish/action.yaml [hunk 1]
```diff
@@ -45,9 +49,13 @@ runs:
...
-        echo "${{ inputs.gcloud-service-account }}" | base64 -d > "${{ inputs.working-directory }}"/google_application_credentials.json
+        if test -n "${{ inputs.gcloud-service-account }}"; then
+          echo "${{ inputs.gcloud-service-account }}" | base64 -d > "${{ inputs.working-directory }}"/google_application_credentials.json
+          GOOGLE_APPLICATION_CREDENTIALS=/app/google_application_credentials.json
+        fi  
+
         docker run \
           --rm \
-          -e GOOGLE_APPLICATION_CREDENTIALS=/app/google_application_credentials.json \
+          -e GOOGLE_APPLICATION_CREDENTIALS=$GOOGLE_APPLICATION_CREDENTIALS \
```
Comment (IuryAlves) [inline techdocs-publish/action.yaml:59]: `GOOGLE_APPLICATION_CREDENTIALS` is already defined when using WIF.

**Review by ronanbarrett — COMMENTED** [review]
Could you use https://github.com/google-github-actions/auth instead of the docker login commands?

**Comment by IuryAlves** [comment]
> Could you use https://github.com/google-github-actions/auth instead of the docker login commands?

I didn't understand the question.

The [google-github-actions/auth docs](https://github.com/google-github-actions/auth#authenticating-to-container-registry-and-artifact-registry)  says that we should do ` docker login -u oauth2accesstoken --password-stdin https://eu.gcr.io`

**Review by K-Phoen — APPROVED** [review]
Did you try to run this new version?

**Comment by IuryAlves** [comment]
> Did you try to run this new version?

Not yet, but I will try it in the Dashung repo.

**Comment by IuryAlves** [comment]
> Did you try to run this new version?

Looks good now: https://github.com/voiapp/dashung/actions/runs/2345933731

**Review by K-Phoen — APPROVED** [review]

**Pull request merged**